src/mlpack/core/stb/stb.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * @file core/data/stb.hpp
+ * @author Omar Shrit
+ *
+ * Header to include stb in mlpack, in addition to allow the user to disable
+ * all of these includes.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_STB_STB_HPP
+#define MLPACK_CORE_STB_STB_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#ifndef STB_IMAGE_RESIZE_IMPLEMENTATION
+#define STB_IMAGE_RESIZE_IMPLEMENTATION
+#include "stb_image_resize.h"
```

Comment:
Maybe it is better to push the handling of `MLPACK_NO_STB` to here?  Then if `MLPACK_NO_STB` is defined, we include the system `stb_image_resize` here.  (Could use the old `__has_include` code here.)

Reply:
I need to think about it because the current idea is that his entire file is not going to be defined if the user includes their own STB. For now, it does not work like this, and I want an easy solution.
Possibly we can move the _has_include here as you suggest.